### ISSUE TITLE
locale is `undefined` in custom 500.js page at getStaticProps

### ISSUE THREAD

**Issue by yfhui** [vercel/next.js#32309]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

16.9.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

local

### Describe the Bug

Hi, I am facing an issue about i18n in a custom 500.js page. The locale returned from the context in `getStaticProps` is `undefined`.

As I am using next-i18next, and I asked the repo owner [there](https://github.com/isaachinman/next-i18next/issues/1307). But he said it should be a nextjs issue. We focus on the `locale` from context of `getStaticProps`

```js
import { useRouter } from 'next/router'

export default function Custom500(props) {
    console.log("rendering react 500 page")
    const { locale } = useRouter();
    console.log("client locale",locale);

    return (
        <div>'error'</div>
    );
}

export const getStaticProps = async ({ req, res, query, locale }) => {
    console.log("rendering 500",locale)
    return { props: {} };
}
```

Here is my notes:
* I intentionally wrote a buggy line to trigger an error
* 500.js or _error.js both have the same issue
* The error page(500.js, _error.js) was rendered at `development` environment, it is not an issue, but:
     - considering we have home page(without err) and 2nd page(with err), home page has a link to 2nd page
     - because nextjs uses the tactics of pregeneration, accessing home page, then will prerender 2nd page, when accessing 2nd page from home page using the link, the generated error page will be served directly
     -![image](https://user-images.githubusercontent.com/48435910/145353051-d31b1066-21c5-431d-a8a7-9d124fe33f5f.png)
        
     - If entering the 2nd page directly from url, then weird things happened:
         * When requesting 2nd page, it encounted the error, then render the error page
         * **The error page is rendered twice, first time it gets the locale defined, but the in the 2nd rendering, locale is undefined.** (also note that the component/react level rendering log disappears)
          ![image](https://user-images.githubusercontent.com/48435910/145353124-d4bfa3ef-5913-435e-a81f-e60a74fae20b.png)
         
* I observe that with the error page is built, but in development env, actaully the real development error page is served, not the 500 page, it should be an intented behavior in nextjs.
    ![image](https://user-images.githubusercontent.com/48435910/145347423-b01fdb5c-12f6-45f8-b5bd-c188b981b567.png)
* As I am just printing `locale` in the example so it doesn't trigger another error, but if I used the locale info for my real 500.js page, then it will throw the undefined error, instead of the real error, making the original error hard to chase
     

* In `production`, locale is well defined, because getStaticProps will generate the page on build
* Althought I tested with the next-18next example, but the locale should be returned by nextjs, so it should be a nextjs issue, not next-i18next issue
* I could also see that [#18513](https://github.com/vercel/next.js/pull/18513) addresses a similiar issue at 404 page. Is there any relationship to that pull request?
* access it directly using localhost:3000/500 so page 500 doesn't have any error

I am thinking even the error page is rendered twice, locale actually should be defined at the 2nd time. So either solving the error page rendering twice issue or check why locale is undefined in `development` envrionment will be a good start of such an issue

**TL;DR: Directly entering a page(with err) using URL will trigger rendering of error page in development env TWICE, the 2nd time of the rendering locale is undefined**

### Expected Behavior

* direct enter of URL will trigger the rendering of the error page, which matches with the current behavior.
* **BUT**
    * we expected it should render once
    * the locale should be always defined

### To Reproduce

To reproduce, you can use check this [repo](https://github.com/yfhui/next-i18next-error-demo)

I have also tested on next@11, the issue presists.

**Comment by yfhui** [comment]
It maybe a duplicate of [#31771](https://github.com/vercel/next.js/issues/31771) but this issue provides a more detailed observation.

* directly enter the URL will trigger error page render twice
* 2nd time will have the locale undefined

**Comment by and-poulsen** [comment]
Did you figure out anything @yfhui , I'm experiencing the same

**Comment by yfhui** [comment]
at the end I changed to use next-translate package, and seems that one doesn't have this issues. But that package is not compatible with i18n package
